external/mcpclient/src/mcp-client-plugin.ts

```diff
@@ -199,10 +201,11 @@ export class McpClientPlugin implements ChatPromptPlugin<'mcpClient', McpClientP
 
   private async fetchTools(
     url: string,
-    headers?: ValueOrFactory<Record<string, string>>,
-    skipIfUnavailable?: boolean
+    headers: ValueOrFactory<Record<string, string>> | undefined,
```

Comment:
what's the difference between "`| undefined`" and "`headers?`"?

Reply:
the need to explicitly set the property to undefined vs it being optional to set.